src/aliceVision/system/hardwareContext.hpp

```diff
@@ -0,0 +1,46 @@
+// This file is part of the AliceVision project.
+// Copyright (c) 2016 AliceVision contributors.
+// This Source Code Form is subject to the terms of the Mozilla Public License,
+// v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Logger.hpp"
+#include "Timer.hpp"
+
+#include <boost/program_options/options_description.hpp>
```

Comment:
Maybe a forward declaration to avoid the dependency of HardwareContext to program_options (or do the job in cmdline).